pilot/pkg/networking/core/v1alpha3/configgen.go

```diff
@@ -96,11 +96,19 @@ func (configgen *ConfigGeneratorImpl) buildSharedPushStateForSidecars(env *model
 				// here will be the default CDS output for a given namespace based on public/private
 				// services and destination rules
 				sc.XDSOutboundClusters = configgen.buildOutboundClusters(env, &dummyNode, push)
+
+				for _, p := range configgen.Plugins {
+					perRouteFilterConfig := p.OnPreComputePerRouteFilterConfig(env, &dummyNode, push)
+					if perRouteFilterConfig != nil {
+						sc.PerRouteFilterConfig = make(map[string]interface{})
+						sc.PerRouteFilterConfig[p.GetName()] = perRouteFilterConfig
+						break
```

Comment:
why break? just allocate the map outside the forloop, and assign to the cache. If in future other plugins have per route filter config, they could use this cache as well

Reply:
DOne~